Miltank rework + Inexorable re-code

zinniagamedev opened the pull request:
- Cut Rough and Tumble (miltank signature move)
- Cut Milk Drink
- Tweaked/buffed Miltank's stats to make it a dedicated Fast Wall (with high HP for Wish purposes)
- Stability (NVSC immunity) replaces Break Through on Miltank
- New ability "Refreshing" replaces Thick Fat on Miltank: "This Pokémon's moves heal for 25% more."
  - Added code support for the healing helper methods being able to care about the user of the move that caused the healing (when relevant). Draining type effects are considered to be "from the user's move" but effects placed by moves (e.g. Aqua Ring) are not. Shell Bell, despite being not technically draining (it doesn't synergize with Big Root, Gorging, etc.) is still counted as "healing from moves" for this purpose.
  - For Wish, I refactored Foretold Moves and Wish to both use a new helper method of the Battle class called "getBattlerFromFieldOrParty". That way, Wish healing can benefit from Refreshing even when the user has swapped out.
- Added new signature move Dinner Bell: "The user calls its team to a farm-fresh meal, sacrificing 25% HP to heal each of its party members by 25% HP."
  - The Aid Kit party heal animation plays when using Dinner Bell It doesn't show any status conditions since they aren't relevant, and it makes sure not to show the healing from any team members that are on the field since that's already covered by the standard healing animation.
- Refactored the move codes for Wish and Arc of Hope so that Arc of Hope now inherits from Wish.
- Break Through replaces Filthy on Rattata line
- Replaced Miltank on Yezera 3's team with Raticate
- Re-coded Inexorable to fix multiple bugs with it. additionally, made the following UX improvements:
  - The ability splash appears for Inexorable when statuses are going off
  - The status animation does not play for the party mons that are being damaged by inexorable status
  - There is now a message when a Pokemon in the party faints from the inexorable status damage

zinniagamedev committed:
Miltank rework pt.1

Cut Rough and Tumble (miltank signature move)
Cut Milk Drink
Tweaked/buffed Miltank's stats
Added new signature move Dinner Bell: "The user calls its team to a farm-fresh meal, sacrificing 25% HP to heal each of its party members by 25% HP."
Haven't actually made the move code for DINNERBELL, and haven't swapped out any abilities

zinniagamedev committed:
Miltank rework pt.2

Reworked Miltank's stats again to make it more of a dedicated Fast Wall (with high HP for Wish purposes)
Stability replaces Breakthrough on it

Added new ability "Refreshing": "This Pokémon's moves heal for 25% more."
Added code support for the healing helper methods being able to care about the user of the move that caused the healing (when relevant). Draining type effects are considered to be "from the user's move" but effects placed by moves (e.g. Aqua Ring) are not. Shell Bell, despite being not technically draining (it doesn't synergize with Big Root, Gorging, etc.) is still counted as "healing from moves" for this purpose.
For Wish, I refactored Foretold Moves and Wish to both use a new helper method of the Battle class called getUserForDelayedMove. That way, Wish healing can benefit from Refreshing even when the user has swapped out.
Also refactored the move codes for Wish and Arc of Hope so that Arc of Hope now inherits from Wish.

zinniagamedev committed:
Actually implement move code for DINNERBELL

Both it and Refreshing need a lot of testing, have only done Auto-Battler testing to check for crashes introduced by the refactors

zinniagamedev committed:
Miltank rework pt. 3

The Aid Kit party heal animation plays when using Dinner Bell It doesn't show any status conditions since they aren't relevant, and it makes sure not to show the healing from any team members that are on the field since that's already covered by the standard healing animation.
Fixed the effect scoring for Dinner Bell
Dinner Bell now fails when there are no team members to heal
Also includes compilation-based changes

zinniagamedev committed:
Break Through replaces Filthy on Rattata line

zinniagamedev committed:
Break Through ability tweaked: Also boosts move damage by 20%

AlphaKretin commented on Plugins/Tectonic Battle/Battle/PokeBattle_Battle.rb:
```diff
@@ -455,4 +455,25 @@ def pbThisEx(idxBattler, idxParty)
+        moveUser = nil
+        eachBattler do |b|
+            next if b.opposes?(userIndex)
+            next if b.pokemonIndex != partyIndex
```
My understanding is that this is supposed to check that the Pokemon it's checking is in your party (as opposed to the opponent's). The variable being called pokemonIndex makes that unclear though. Am I wrong, is something here wrong, or is it just a poorly named variable?

zinniagamedev replied:
Just a poorly named variable, yeah.

zinniagamedev replied:
It's checking whether or not that battler is the delayed move user that they're looking for (e.g. the Future Sight user is active in the battle, still.)

Azeler3 commented on Plugins/Tectonic Battle/Move/Move_Usage_Calculations.rb:
```diff
@@ -395,7 +395,7 @@ def canApplyRandomAddedEffects?(user, target, chance, showMessages = false, aiCh
 
     def pbAdditionalEffectChance(user, target, type, effectChance = 0, aiCheck = false)
-        return 100 if @battle.pbCheckGlobalAbility(:WISHMAKER)
+        return 100 if @battle.pbCheckGlobalAbility(:WishMakerPartyIndex)
```
Find and Replace typo. WISHMAKER here refers to the ability.

zinniagamedev replied:
Good find, thanks!

zinniagamedev committed:
Fix accidentally breaking WISHMAKER ability wish Wish related code changes

zinniagamedev committed:
Rename getUserForDelayedMove to getBattlerFromFieldOrParty

zinniagamedev committed:
Inexorable re-code and UX improvements

The ability splash appears for Inexorable when statuses are going off
The status animation does not play for the party mons that are being damaged by inexorable status
There is now a message when a Pokemon in the party faints from the inexorable status damage

zinniagamedev committed:
Replace Miltank on Yezera 3's team with Raticate
